showCrime/dailyIncid/management/commands/parse_UCR.py

```diff
@@ -0,0 +1,397 @@
+
+	allKeys = list(allStats.keys())
+	allKeys.sort()
+	
+	# allLblTbl: lbl -> date -> div -> freq
+	allLblTbl = defaultdict(lambda: defaultdict(lambda: defaultdict(int)))
+	for k in allKeys:
+		divs,rptDate = k.split('_')
+		div = int(divs)
+		
+		# NB: need to distinguish stat's lbl which may have \n from mlbl used in LabelUCROrder
+		for slbl in allStats[k]:
+			if slbl.find('\n') != -1:
+				mlbl = slbl.split('\n')[0]
+			else:
+				mlbl = slbl
+			if mlbl in LabelUCROrder:
+				allLblTbl[mlbl][rptDate][div] =  allStats[k][slbl]
+			elif mlbl not in IgnoreStatLbl:
+				print("combineWeeksUCR: unknown label?! %s %s" % (mlbl,k))
+				
+	return allLblTbl
+
+def rptAllStats(allLblTbl,outf):
+	"""produce CSV file of UCR crime categories X dates, breaking out indiv divisions' subtotals
+	"""
+	allDatesSet = set()
+	for lbl in allLblTbl.keys():
+		allDatesSet.update(allLblTbl[lbl].keys())
+	allDates = list(allDatesSet)
+	allDates.sort()
+
+	# ASSUME five OPD division
+	OPDivs = range(1,6)
+	outs = open(outf,'w')
+
+	line = '"UCRCategory \ Date"'
+	for dates in allDates:
+		dateDup = 6 * (','+dates)
+		line += ('%s' % (dateDup))
+	outs.write(line+'\n')
+	
+	line = 'Division'
+	for dates in allDates:
+		for div in OPDivs:
+			line += (',%d' % (div))
+		line += (',tot')
+	outs.write(line+'\n')
+
+	for lbl in LabelUCROrder:
+		albl = lbl.encode(encoding='utf_8', errors='strict').decode('ascii','ignore')
+		line = '"%s"' % (albl)
+		for dates in allDates:
+			tot = 0
+			for div in OPDivs:
+				if dates in allLblTbl[lbl] and div in allLblTbl[lbl][dates]:
+					val = int(allLblTbl[lbl][dates][div])
+				else:
+					val = 0
+				line += (',%d' % (val))
+				tot += val
+			line += (',%d' % (tot))
+		line += '\n'
+		outs.write(line)
+
+	outs.close()
+
+def rptSummStats(allLblTbl,outf):
+	"""produce CSV file of UCR crime categories X dates, summing divisions' subtotals
+	also echo crime categories across all data
+	"""
+	allDatesSet = set()
+	for lbl in allLblTbl.keys():
+		allDatesSet.update(allLblTbl[lbl].keys())
+	allDates = list(allDatesSet)
+	allDates.sort()
+
+	lblTots = defaultdict(int)
+	
+	# ASSUME five OPD division
+	OPDivs = range(1,6)
+	outs = open(outf,'w')
+
+	line = '"UCRCategory \ Date"'
+	for dates in allDates:
+		line += (',%s' % (dates))
+	outs.write(line+'\n')
+	for lbl in LabelUCROrder:
+		albl = lbl.encode(encoding='utf_8', errors='strict').decode('ascii','ignore')
+		line = '"%s"' % (albl)
+		for dates in allDates:
+			tot = 0
+			for div in OPDivs:
+				if dates in allLblTbl[lbl] and div in allLblTbl[lbl][dates]:
+					tot += int(allLblTbl[lbl][dates][div])
+			lblTots[lbl] += tot
+			line += (',%s' % (tot))
+		line += '\n'
+		outs.write(line)
+	outs.close()
+	
+	for lbl in LabelUCROrder:
+		if lblTots[lbl] == 0:
+			print('%s\t%d <=======' % (lbl,lblTots[lbl]))
+		else:	
+			print('%s\t%d' % (lbl,lblTots[lbl]))
+
+# Police Area 3 Weekly Crime Reports
+DivDirName_pat = re.compile(r'Police Area (\d) Weekly Crime Reports')
```

Comment:
Sorry I took a while to review - I was trying to figure out why I wasn't getting any stats. Turned out my folder and file names weren't matching the regex. I got it to work, and it looks good to me. Maybe we can define these as a configuration at a later point, so we don't have to hardcode it and change the code in case the file names change?